bin/faucet/src/server/get_tokens.rs

```diff
@@ -178,13 +174,12 @@ impl RawMintRequest {
             return Err(InvalidRequest::InvalidApiKey(api_key.clone()));
         }
 
-        if let Ok(pow_parameters) = PowParameters::try_from(&self) {
-            pow_parameters.check_pow_solution(&server.pow.challenge_cache)?;
-            pow_parameters.check_server_timestamp()?;
-            pow_parameters.check_server_signature(&server.pow.salt)?;
-        } else {
-            return Err(InvalidRequest::MissingPowParameters);
-        }
+        // Validate Challenge and nonce
+        let challenge_str = self.challenge.ok_or(InvalidRequest::MissingPowParameters)?;
+        let nonce = self.nonce.ok_or(InvalidRequest::MissingPowParameters)?;
+
+        let challenge = Challenge::decode(&challenge_str, server.pow.salt)?;
+        server.pow.submit_challenge(&challenge, nonce)?;
```

Comment:
nit: I would maybe maybe have `submit_challenge()` method on the `server` which would take challenge string, decode it, and then call `submit_challenge()` on the internal `pow` instance.

Reply:
I don't think that this decoupling is right because it would cause fragmentation of the context. wdyt?

Reply:
What I'm trying to achieve is to make `pow` an internal concern of the `server`. That is, ideally, `pow` would be a private field of the `server` struct and we won't be accessing it directly.